### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by Gerfey**

### CODE REVIEW

File: cmd/shortener/flag.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"flag"
+	"os"
+)
+
+var (
+	flagRunServerAddress       string
```
Comment (arttet) [inline cmd/shortener/flag.go:9]: [CHANGE REQUEST] Please don't use global variables

File: cmd/shortener/main.go [hunk 2]
```diff
@@ -1,3 +1,25 @@
 package main
 
-func main() {}
+import (
+	"github.com/Gerfey/shortener/internal/app/config"
+	"github.com/Gerfey/shortener/internal/pkg/app"
+	"log"
```
Comment (arttet) [inline cmd/shortener/main.go:6]: [CHANGE REQUEST] Please sort imports
https://stackoverflow.com/questions/38704769/golang-import-grouping-by-package

File: internal/app/generator/generator.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package generator
+
+import "math/rand"
+
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+type Generator interface {
```
Comment (arttet) [inline internal/app/generator/generator.go:7]: [CLARIFICATION] Why do you need it?

File: internal/app/store/store.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package store
+
+import "sync"
+
+type Store struct {
```
Comment (arttet) [inline internal/app/store/store.go:5]: [CLARIFICATION] Why don't you use `interface`?

File: internal/pkg/app/app.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	"github.com/Gerfey/shortener/internal/app/generator"
+	"github.com/Gerfey/shortener/internal/app/store"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"log"
+	"net/http"
+)
+
+type App struct {
+	c *config.Config
```
Comment (arttet) [inline internal/pkg/app/app.go:15]: [CHANGE REQUEST] Please rename it.
https://github.com/uber-go/guide/blob/master/style.md

**Review by arttet — CHANGES_REQUESTED** [review]
- [ ] Split the application into layers
https://github.com/golang-standards/project-layout
https://medium.com/@deanrubin/the-three-layered-architecture-fe30cb0e4a6#:~:text=The%20presentation%20layer%20is%20the,Update%20or%20Delete)%20operation%20flow.
Please integrate CodeCov https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- [ ] Please read and use the Conventional Commits specification https://www.conventionalcommits.org/en/v1.0.0/
- [ ] Please integrate the linter https://github.com/golangci/golangci-lint-action
- [ ] Please don't use global variables
- [ ] Please delete deprecated constraints in CI
https://github.com/Gerfey/shortener/blob/e90b90d0a3e871d0c4f9109ffb1b2c445448912c/.github/workflows/shortenertest.yml#L65

File: cmd/shortener/flag.go (hunk 1 above)
Comment (Gerfey) [reply cmd/shortener/flag.go:9]: @arttet в курсе одно, к код ревью другое?)

File: cmd/shortener/main.go (hunk 2 above)
Comment (Gerfey) [reply cmd/shortener/main.go:6]: Пакеты сортируются автоматом GoLand

File: internal/app/generator/generator.go (hunk 3 above)
Comment (Gerfey) [reply internal/app/generator/generator.go:7]: Это генератор, который будет создавать из случайной последовательности код для сокращения URL

File: .github/workflows/codecov.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+          go-version: 1.22
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Run tests with coverage
+        run: go test ./... -coverprofile=coverage.out
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```
Comment (arttet) [inline .github/workflows/codecov.yml:32]:
```suggestion
        uses: codecov/codecov-action@v4
```

Comment (arttet) [reply .github/workflows/codecov.yml:32]: https://github.com/codecov/codecov-action

Comment (arttet) [inline .github/workflows/codecov.yml:32]: [SUGGESTION] You can use Dependabot

https://gist.github.com/magnetikonline/6f215db058e327905bce66c37f92426c

https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot

File: internal/app/repository/repository.go [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package repository
+
+type Repository interface {
+	Find(key string) (string, bool)
```
Comment (arttet) [inline internal/app/repository/repository.go:4]:
```suggestion
	FindURL(key string) (string, bool)
```

File: internal/pkg/app/app.go [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	log.Println("Starting server...")
+
+	a.router.Route("/", func(r chi.Router) {
+		r.Post("/", a.handler.ShortenURLHandler)
+		r.Get("/{id}", a.handler.RedirectURLHandler)
+	})
+
+	err := http.ListenAndServe(a.settings.ServerAddress(), a.router)
+	if err != nil {
+		log.Fatal(err)
```
Comment (arttet) [inline internal/pkg/app/app.go:52]: [SUGGESTION] It would be better if you returned an error.

**Review by arttet — APPROVED** [review]

**Pull request merged**